### ISSUE TITLE
Facing missing type id property '_class' while serializing

### ISSUE THREAD

**Issue by nomadus** [spring-projects/spring-statemachine#637]
I am trying to create REST methods based on SM service methods, one of them is as follows
```
  @RequestMapping(value = "",
            method = RequestMethod.POST,
            consumes = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<String> createEngine(
            @RequestBody JpaRepositoryStateMachine engine) {
             newEngine = new JpaRepositoryStateMachine();         
            newEngine.setState(engine.getState());           
            engineServiceImpl.createEngine(newEngine);
}
```
And on line which has @RequestBody JpaRepositoryStateMachine engine, it throws:
 "message": "JSON parse error: Missing type id when trying to resolve subtype of [simple type, class org.springframework.statemachine.data.jpa.JpaRepositoryStateMachine]: missing type id property '_class'; nested exception is com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Missing type id when trying to resolve subtype of [simple type, class org.springframework.statemachine.data.jpa.JpaRepositoryStateMachine]: missing type id property '_class'\n at [Source: (PushbackInputStream); line: 4, column: 1]",
    "path": "/engine"

It can be fixed if I send "_class" attribute in the request body, but this should not work this way.

{
    "_class": "org.springframework.statemachine.data.jpa.JpaRepositoryStateMachine",
    "machineId": "mymachine",
    "state": "SUBMITTED",
    "stateMachineContext": null
}

**Comment by jvalkeal** [comment]
Yeah that's indeed required for now and is used like that i.e. in samples where data is populated when app starts. It indeed comes from a base entity class

https://github.com/spring-projects/spring-statemachine/blob/f67ec1380bafa625c9a6840bbf986a8b29a3d399/spring-statemachine-data/src/main/java/org/springframework/statemachine/data/BaseRepositoryEntity.java#L26-L28

I wonder what would be a best way to _fix_ this to make functionality more sane?
